codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -57,15 +58,30 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.remoteFlowState(new DefaultFlowState(stream, 0));
+                // If the stream state is not open then the stream is not yet eligible for flow controlled frames and
+                // only requires the ReducedFlowState. Otherwise the full amount of memory is required.
+                stream.remoteFlowState(stream.state() == IDLE ?
```

Comment:
@nmittler - See https://github.com/netty/netty/pull/3677#issuecomment-94935569. This demonstrates the savings for streams created in IDLE.  This doesn't account for any savings we will see from reclaiming memory for CLOSED streams.